File: app/Models/Votante.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class Votante extends Model
{
    use HasFactory;

    protected $connection = 'sqlsrv_db1';
    protected $table = 'votantes';

    public $timestamps = false;

    protected $fillable = [
        'cedula',
        'dirigente_id',
        'colegio_id',
        'mesa',
        'telefono',
        'direccion',
        'latitud',
        'longitud',
        'direccion_gps',
        'gps_capturado_en',
        'observaciones',
        'fecha_registro',
        'activo'
    ];

    protected $casts = [
        'activo' => 'boolean',
        'fecha_registro' => 'datetime',
        'gps_capturado_en' => 'datetime',
        'latitud' => 'float',
        'longitud' => 'float'
    ];

    protected $appends = ['padron_data', 'colegio_nombre'];

    public function dirigente()
    {
        return $this->belongsTo(User::class, 'dirigente_id');
    }

    public function gestiones()
    {
        return $this->hasMany(GestionVotante::class, 'votante_id');
    }

    public function getPadronDataAttribute()
    {
        $padron = \DB::connection('sqlsrv_db1')
            ->table('Padron as p')
            ->leftJoin('Nacionalidad as nac', 'p.IdNacionalidad', '=', 'nac.ID')
            ->leftJoin('Sexo as sex', 'p.IdSexo', '=', 'sex.IdSexo')
            ->leftJoin('EstadoCivil as ec', 'p.IdEstadoCivil', '=', 'ec.Id')
            ->leftJoin('Colegio as col', 'p.IdColegio', '=', 'col.IDColegio')
            ->leftJoin('Municipio as mun', 'p.IdMunicipio', '=', 'mun.ID')
            ->leftJoin('Provincia as pro', 'mun.IDProvincia', '=', 'pro.ID')
            ->where('p.Cedula', $this->cedula)
            ->select([
                'p.Cedula',
                'p.nombres',
                'p.apellido1',
                'p.apellido2',
                'p.FechaNacimiento',
                'nac.Descripcion as nacionalidad',
                'sex.Descripcion as sexo',
                'ec.Descripcion as estado_civil',
                'col.Descripcion as colegio_nombre',
                'mun.Descripcion as municipio',
                'pro.Descripcion as provincia',
            ])
            ->first();

        return $padron;
    }

    public function getColegioNombreAttribute()
    {
        if (!$this->colegio_id) return null;

        $colegio = \DB::connection('sqlsrv_db1')
            ->table('Colegio')
            ->where('IDColegio', $this->colegio_id)
            ->value('Descripcion');

        return $colegio;
    }

    public function getFotoDataAttribute()
    {
        try {
            $foto = \DB::connection('sqlsrv_db2')
                ->table('FOTOS_PRM_PRM')
                ->where('Cedula', $this->cedula)
                ->select('Imagen')
                ->first();

            return $foto;
        } catch (\Exception $e) {
            return null;
        }
    }

    public function scopeDelDirigente($query, $dirigenteId)
    {
        return $query->where('dirigente_id', $dirigenteId);
    }

    public function scopeActivos($query)
    {
        return $query->where('activo', true);
    }

    public function scopePorColegio($query, $colegioId)
    {
        return $query->where('colegio_id', $colegioId);
    }

    public function scopePorMesa($query, $mesa)
    {
        return $query->where('mesa', $mesa);
    }
}

File: flutter_votantes_app/lib/models/votante.dart
class Votante {
  final int? id;
  final String cedula;
  final int dirigenteId;
  final int? colegioId;
  final String? mesa;
  final String? telefono;
  final String? direccion;
  final String? observaciones;
  final DateTime? fechaRegistro;
  final bool activo;
  final PadronData? padronData;
  final String? colegioNombre;

  Votante({
    this.id,
    required this.cedula,
    required this.dirigenteId,
    this.colegioId,
    this.mesa,
    this.telefono,
    this.direccion,
    this.observaciones,
    this.fechaRegistro,
    this.activo = true,
    this.padronData,
    this.colegioNombre,
  });

  factory Votante.fromJson(Map<String, dynamic> json) {
    return Votante(
      id: json['id'],
      cedula: json['cedula'],
      dirigenteId: json['dirigente_id'],
      colegioId: json['colegio_id'],
      mesa: json['mesa'],
      telefono: json['telefono'],
      direccion: json['direccion'],
      observaciones: json['observaciones'],
      fechaRegistro: json['fecha_registro'] != null
          ? DateTime.parse(json['fecha_registro'])
          : null,
      activo: json['activo'] ?? true,
      padronData: json['padron_data'] != null
          ? PadronData.fromJson(json['padron_data'])
          : null,
      colegioNombre: json['colegio_nombre'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      if (id != null) 'id': id,
      'cedula': cedula,
      'dirigente_id': dirigenteId,
      'colegio_id': colegioId,
      'mesa': mesa,
      'telefono': telefono,
      'direccion': direccion,
      'observaciones': observaciones,
      'activo': activo,
    };
  }

  String get nombreCompleto {
    if (padronData != null) {
      return '${padronData!.nombres} ${padronData!.apellido1} ${padronData!.apellido2}'.trim();
    }
    return 'Sin nombre';
  }
}

class PadronData {
  final String cedula;
  final String nombres;
  final String? apellido1;
  final String? apellido2;
  final DateTime? fechaNacimiento;
  final String? lugarNacimiento;
  final String? nacionalidad;
  final String? sexo;
  final String? estadoCivil;
  final String? colegio;
  final int? colegioId;
  final String? municipio;
  final String? provincia;

  PadronData({
    required this.cedula,
    required this.nombres,
    this.apellido1,
    this.apellido2,
    this.fechaNacimiento,
    this.lugarNacimiento,
    this.nacionalidad,
    this.sexo,
    this.estadoCivil,
    this.colegio,
    this.colegioId,
    this.municipio,
    this.provincia,
  });

  factory PadronData.fromJson(Map<String, dynamic> json) {
    return PadronData(
      cedula: json['Cedula'],
      nombres: json['Nombres'],
      apellido1: json['Apellido1'],
      apellido2: json['Apellido2'],
      fechaNacimiento: json['FechaNacimiento'] != null
          ? DateTime.parse(json['FechaNacimiento'])
          : null,
      lugarNacimiento: json['LugarNacimiento'],
      nacionalidad: json['nacionalidad'],
      sexo: json['sexo'],
      estadoCivil: json['estado_civil'],
      colegio: json['colegio'],
      colegioId: json['colegio_id'],
      municipio: json['municipio'],
      provincia: json['provincia'],
    );
  }
}